java/README.md

```diff
@@ -0,0 +1,201 @@
+# Velox4J: Java Bindings for Velox
+
+![badge](https://github.com/facebookincubator/velox/actions/workflows/java-ut-java.yml/badge.svg?branch=main)
+![badge](https://github.com/facebookincubator/velox/actions/workflows/java-ut-cpp.yml/badge.svg?branch=main)
+
+## Introduction
+
+### What is Velox4J?
+
+Velox4J is the Java bindings for Velox. It enables JVM applications to directly invoke Velox's
+functionalities without writing and maintaining any C++ / JNI code.
+
+## Design
+
+Velox4J is designed within the following manners:
+
+### Portable
+
+Velox4J is designed to be portable. The eventual goal is to make one Velox4J release to be
+shipped onto difference platforms without rebuilding the Jar file.
+
+### Seamless Velox API Mapping
+
+Velox4J directly adopts Velox's existing JSON serde framework and implements the following
+JSON-serializable Velox components in Java-side:
+
+- Data types
+- Query plans
+- Expressions
+- Connectors
+
+With the help of Velox's own JSON serde, there will be no re-interpreting layer for query plans
+in Velox4J's C++ code base. Which means, the Java side Velox components defined in Velox4J's
+Java code will be 1-on-1 mapped to Velox's associated components. The design makes Velox4J's
+code base even small, and any new Velox features easy to add to Velox4J.
+
+### Compatible With Arrow Java
+
+Velox4J is compatible with [Apache Arrow's Java implementation](https://arrow.apache.org/java/). Built-in utilities converting between
+Velox4J's RowVector / BaseVector and Arrow Java's VectorSchemaRoot / Table / FieldVector are provided.
+
+## Prerequisites
+
+### Platform 
+
+The project and its releases are now only tested on the following CPU architectures:
+
+- x86-64
+
+and on the following operating systems:
+
+- Linux
+
+Supports for platforms not on the above list will not be guaranteed to have by the main stream code
+of Velox4J at the time. But certainly, contributions are always welcomed if anyone tends to involve.
+
+### Build Toolchains
+
+The minimum toolchain versions for building Velox4J:
+
+- GCC 11
+- JDK 11
```

Comment:
Are there plans to support JDK 8, considering that some customers might still be using it?

Reply:
It's a valid point to me. And I think it's not very difficult to add the support given we don't use much of Java 11 language features yet. Could be a topic for following up.